addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -466,13 +466,20 @@ def get_allocation_data(self, employees, target_date=None):
                     closest_allocation_remaining += allocations_leaves_consumed[employee][leave_type][closest_allocation]['virtual_remaining_leaves']
                 if closest_allocation.date_to:
                     closest_allocation_expire = format_date(self.env, closest_allocation.date_to)
+                    calendar = employee.resource_calendar_id or self.env.company.resource_calendar_id
                     # closest_allocation_duration corresponds to the time remaining before the allocation expires
-                    closest_allocation_duration =\
-                        employee.resource_calendar_id._attendance_intervals_batch(
-                            datetime.combine(closest_allocation.date_to, time.min).replace(tzinfo=pytz.UTC),
-                            datetime.combine(target_date, time.max).replace(tzinfo=pytz.UTC))\
-                        if leave_type.request_unit in ['hour']\
-                        else (closest_allocation.date_to - target_date).days + 1
+                    calendar_attendance = calendar._work_intervals_batch(
+                        datetime.combine(target_date, time.min).replace(tzinfo=pytz.UTC),
+                        datetime.combine(closest_allocation.date_to, time.max).replace(tzinfo=pytz.UTC),
+                        resources=employee.resource_id
+                    )
```

Comment:
Note: Args are not in the same order (target_date and closest_allocation)

Is it intended ?